src/ripple/app/consensus/RCLConsensus.cpp

```diff
@@ -735,6 +804,12 @@ RCLConsensus::Adaptor::onModeChange(
 {
     JLOG(j_.info()) << "Consensus mode change before=" << to_string(before)
                     << ", after=" << to_string(after);
+
+    // If we were proposing but aren't any longer, we need to reset the
+    // censorship tracking to avoid bogus warnings.
+    if (before == ConsensusMode::proposing && before != after)
```

Comment:
Now that both validators and non-validators are tracking, I believe this check also needs to change to handle when a non-validator server detects it is working on the wrong ledger:
```suggestion
    if ((before == ConsensusMode::proposing || before == ConsensusMode::observing) && before != after)
```